### PR TITLE
issue #51 : Check valid password length

### PR DESCRIPTION
**Pull request by julielee228**
The password length should be between 6 and 16.
Send an error message if different the scope.

### CODE REVIEW

File: Backend/router/v1/user.js [hunk 1]
```diff
@@ -4,6 +4,9 @@ const models = require('../../models');
 const crypto = require('crypto');
 const session = require('express-session');
 
+let MIN_PASSWORD_LENGTH = 6;
+let MAX_PASSWORD_LENGTH = 16;
```
Comment (shberry) [inline Backend/router/v1/user.js]: Is there a reason we're limiting max to 16? Some of my passwords are longer than that, and I created then like them in purpose for security reasons.

File: Backend/router/v1/user.js [hunk 2]
```diff
@@ -32,14 +35,18 @@ router.post('/signup', (req, res) => {
...
-      });
+      if (req.body.password.length < MIN_PASSWORD_LENGTH || req.body.password.length > MAX_PASSWORD_LENGTH ) {
+        res.status(400).send("Password must be between " + MIN_PASSWORD_LENGTH + ' and ' + MAX_PASSWORD_LENGTH + " characters long");
+      } else {
+        models.user.create({
+          email: req.body.email,
+          password: hashedPassword,
+        }).then(result => {
+            res.send(
+              "< " + req.body.email + " > membership has been completed."
```
Comment (shberry) [inline Backend/router/v1/user.js]: What does 'membership has been completed' mean? I don't think the message is clear.

File: Backend/router/v1/user.js [hunk 3]
```diff
@@ -62,14 +69,18 @@ router.get('/signin', (req, res) => {
     {
       res.json({msg: "User does not exist"});
     } else {
-      let dbPassword = result[0].dataValues.password;
-      let hashPassword = crypto.createHash("sha512").update(req.query.password).digest("hex");
-      if(dbPassword == hashPassword) {
-        req.session.email = req.query.email;
-        res.json({msg: req.session.email});
+      if (req.body.password.length < MIN_PASSWORD_LENGTH || req.body.password.length > MAX_PASSWORD_LENGTH ) {
```
Comment (shberry) [inline Backend/router/v1/user.js]: I see this if statement repeated here and above. Can you refactor this out?

File: Backend/router/v1/user.js (hunk 2 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: Can I change it like "Membership registration is complete" ?

Comment (julielee228) [reply Backend/router/v1/user.js]: Yes, thank you.

File: Backend/router/v1/user.js (hunk 3 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: I will make a password_length_check function.

Comment (shberry) [reply Backend/router/v1/user.js]: No need to update the comment on what you 'will' be doing. Just resolve this once you're actually done :)

File: Backend/router/v1/user.js (hunk 1 above)
Comment (shberry) [reply Backend/router/v1/user.js]: I was talking about max, not min.

File: Backend/router/v1/user.js [hunk 4]
```diff
@@ -26,6 +25,13 @@ router.all('/', (req, res) => {
   res.send("this is user root");
 });
 
+function password_length_check(length){
+  if(length < MIN_PASSWORD_LENGTH) {
```
Comment (shberry) [inline Backend/router/v1/user.js]: These 4 lines can be combined into one. Can you try?

File: Backend/router/v1/user.js (hunk 3 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: Oh, yes!

File: Backend/router/v1/user.js (hunk 4 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: I can do it!

File: Backend/router/v1/user.js (hunk 1 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: I modified password limitation as 8 to infinity

**Pull request merged**